src/ws/shard.ts

```diff
@@ -15,8 +15,9 @@ import {
 import { BotStatusRequest, delay } from "../util/utils.ts";
 import { decompressWith } from "./deps.ts";
 import { handleDiscordPayload } from "./shard_manager.ts";
+import { Collection } from "../util/collection.ts";
 
-const basicShards = new Map<number, BasicShard>();
+const basicShards = new Collection<number, BasicShard>();
```

Comment:
New Collection, why? We don't really need extra methods

Reply:
It's needed for Promise.all at this line: https://github.com/TriForMine/discordeno/blob/70a3eef41aac9407c33e940584f584de8d6f605f/src/ws/shard.ts#L331
I'm not sure if there is an other way to do it.